ms2/src/org/labkey/ms2/Protein.java

```diff
@@ -823,6 +930,39 @@ public Map<String, PeptideCounts> getUniquePeptides(MS2Run run)
         return uniquePeptides;
     }
 
+    public List<Range> getModifiedPeptideRanges(MS2Run run)
+    {
+        List<Range> modifiedPeptides = new ArrayList<>();
+        if (null == _modifiedPeptideCharacteristics || _modifiedPeptideCharacteristics.size() <= 0)
+            return modifiedPeptides;
+
+        // if called from old-style getCoverageRanges, the run value is 0 and we don't care about modifications
+        List<MS2Modification> mods = Collections.emptyList();
+        if (run != null && run.getRun() > -1)
+            mods = MS2Manager.getModifications(run);
+
+        for (PeptideCharacteristic peptide : _modifiedPeptideCharacteristics)
+        {
+            PeptideCounts peptideCounts = new PeptideCounts();

```

Comment:
Can we consolidate copying these properties with the code from lines 915-924 above?